upb/reflection.c

```diff
@@ -373,6 +373,35 @@ bool upb_array_append(upb_array *arr, upb_msgval val, upb_arena *arena) {
   return true;
 }
 
+
+bool upb_array_insert(upb_array *arr, size_t i, size_t count,
+                      upb_arena *arena) {
+  UPB_ASSERT(i <= arr->len);
+  UPB_ASSERT(count + arr->len >= count);
+  size_t oldsize = arr->len;
+  if (!upb_array_resize(arr, arr->len + count, arena)) {
+    return false;
+  }
+  char* data = _upb_array_ptr(arr);
+  int lg2 = arr->data & 7;
+  memmove(&data[(i + count) << lg2], &data[i << lg2], (oldsize - i) << lg2);
```

Comment:
can you poison the existing elements in sanitized mode?

Reply:
That's a good idea, but it would require systematically changing all functions that change the size of the list. Filed https://github.com/protocolbuffers/upb/issues/464